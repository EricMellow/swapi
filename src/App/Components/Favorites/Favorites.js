import React from 'react';
import './Favorites.css';

const Favorites = ({ favorites, toggleFavorite }) => {
  let favoriteCards = [];
  if (favorites) {
    favoriteCards = favorites.map((favorite, index) => {
      let switchCase = favorite.type;
      console.log(switchCase)
      if (switchCase === 'ship') {
        switchCase = (
          <div key={favorite.name + index} className="card">
            <h3>{favorite.name}</h3>
            <button className="fav" onClick={() => toggleFavorite(favorite)}></button>
            <p>Model: {favorite.model}</p>
            <p>Hyperdrive Rating: {favorite.hyperdriveRating}</p>
            <p>Number of Passengers: {favorite.numOfPassengers}</p>
          </div>
        );
      } 
      if (switchCase === 'planet') {
        switchCase = (
          <div key={favorite.name + index} className="card">
            <h3>{favorite.name}</h3>
            <button className="fav" onClick={() => toggleFavorite(favorite)}></button>
            <p>Terrain: {favorite.terrain}</p>
            <p>Climate: {favorite.climate}</p>
            <p>Population: {favorite.population}</p>
            <p>Residents: {favorite.residents}</p>
          </div>
        );
      } 
      if (switchCase === 'vehicle') {
        switchCase = (
          <div key={favorite.name + index} className="card">
            <h3>{favorite.name}</h3>
            <button className="fav" onClick={() => toggleFavorite(favorite)}></button>
            <p>Model: {favorite.model}</p>
            <p>Class: {favorite.vehicleClass}</p>
            <p>Number of Passengers: {favorite.numOfPassengers}</p>
          </div>
        );
      } 
      if (switchCase === 'person') {
        switchCase = (
          <div key={favorite.name + index} className="card">
            <h3>{favorite.name}</h3>
            <button className="fav" onClick={() => toggleFavorite(favorite)}></button>
            <p>Species: {favorite.species}</p>
            <p>Homeworld: {favorite.homeworld}</p>
            <p>Homeworld Population: {favorite.population}</p>
          </div>
        );
      }
      return switchCase;
    });
  }

  return (
    <div className="container">
      {favoriteCards.length === 0 && <div className="wait">Access code YT-1300. No favorites found.</div>}
      {favoriteCards.length > 0 && favoriteCards}
    </div>
  );
};

export default Favorites;