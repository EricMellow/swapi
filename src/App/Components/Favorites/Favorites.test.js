import React from 'react';
import Favorites from './Favorites';
import { shallow } from 'enzyme';

describe('People', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Favorites />);

  });

  it('should match the snapshot with no data', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot with correct people data', () => {
    const mockData = [{
      id: 'Mr. Test1',
      name: 'Mr Test',
      homeworld: 'Planet',
      species: 'being',
      population: 0,
      type: 'person'
    }];

    wrapper = shallow(<Favorites cardData={mockData} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot with correct planet data', () => {
    const mockData = [{
      id: 'Mr. Planet1',
      name: 'Mr Planet',
      terrain: 'grass',
      population: 200,
      climate: 'nice',
      residents: ['thing1', 'thing2'],
      type: 'planet'
    }];

    wrapper = shallow(<Favorites cardData={mockData} />);

    expect(wrapper).toMatchSnapshot();
  });
  
  it('should match the snapshot with correct starship data', () => {
    const mockData = [{
      id: 'Mr. Ship1',
      name: 'Mr. Ship',
      model: 'Gisele',
      hyperdriveRating: 3,
      numOfPassengers: 2,
      type: 'ship'
    }];

    wrapper = shallow(<Favorites cardData={mockData} />);

    expect(wrapper).toMatchSnapshot();
  });
  it('should match the snapshot with correct vehicle data', () => {
    const mockData = [{
      id: 'Mr. Car1',
      name: 'Mr Car',
      model: "Kate Moss",
      vehicleClass: "Mod 3",
      numOfPassengers: 2,
      type: 'vehicle'
    }];

    wrapper = shallow(<Favorites cardData={mockData} />);

    expect(wrapper).toMatchSnapshot();
  });
});