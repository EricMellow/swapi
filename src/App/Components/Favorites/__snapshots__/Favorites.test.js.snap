// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`People should match the snapshot with correct people data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Favorites
    cardData={
        Array [
            Object {
              "homeworld": "Planet",
              "id": "Mr. Test1",
              "name": "Mr Test",
              "population": 0,
              "species": "being",
              "type": "person",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="wait"
>
          Access code YT-1300. No favorites found.
</div>,
        false,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Access code YT-1300. No favorites found.",
          "className": "wait",
        },
        "ref": null,
        "rendered": "Access code YT-1300. No favorites found.",
        "type": "div",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="wait"
>
            Access code YT-1300. No favorites found.
</div>,
          false,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Access code YT-1300. No favorites found.",
            "className": "wait",
          },
          "ref": null,
          "rendered": "Access code YT-1300. No favorites found.",
          "type": "div",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`People should match the snapshot with correct planet data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Favorites
    cardData={
        Array [
            Object {
              "climate": "nice",
              "id": "Mr. Planet1",
              "name": "Mr Planet",
              "population": 200,
              "residents": Array [
                "thing1",
                "thing2",
              ],
              "terrain": "grass",
              "type": "planet",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="wait"
>
          Access code YT-1300. No favorites found.
</div>,
        false,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Access code YT-1300. No favorites found.",
          "className": "wait",
        },
        "ref": null,
        "rendered": "Access code YT-1300. No favorites found.",
        "type": "div",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="wait"
>
            Access code YT-1300. No favorites found.
</div>,
          false,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Access code YT-1300. No favorites found.",
            "className": "wait",
          },
          "ref": null,
          "rendered": "Access code YT-1300. No favorites found.",
          "type": "div",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`People should match the snapshot with correct starship data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Favorites
    cardData={
        Array [
            Object {
              "hyperdriveRating": 3,
              "id": "Mr. Ship1",
              "model": "Gisele",
              "name": "Mr. Ship",
              "numOfPassengers": 2,
              "type": "ship",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="wait"
>
          Access code YT-1300. No favorites found.
</div>,
        false,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Access code YT-1300. No favorites found.",
          "className": "wait",
        },
        "ref": null,
        "rendered": "Access code YT-1300. No favorites found.",
        "type": "div",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="wait"
>
            Access code YT-1300. No favorites found.
</div>,
          false,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Access code YT-1300. No favorites found.",
            "className": "wait",
          },
          "ref": null,
          "rendered": "Access code YT-1300. No favorites found.",
          "type": "div",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`People should match the snapshot with correct vehicle data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Favorites
    cardData={
        Array [
            Object {
              "id": "Mr. Car1",
              "model": "Kate Moss",
              "name": "Mr Car",
              "numOfPassengers": 2,
              "type": "vehicle",
              "vehicleClass": "Mod 3",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="wait"
>
          Access code YT-1300. No favorites found.
</div>,
        false,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Access code YT-1300. No favorites found.",
          "className": "wait",
        },
        "ref": null,
        "rendered": "Access code YT-1300. No favorites found.",
        "type": "div",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="wait"
>
            Access code YT-1300. No favorites found.
</div>,
          false,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Access code YT-1300. No favorites found.",
            "className": "wait",
          },
          "ref": null,
          "rendered": "Access code YT-1300. No favorites found.",
          "type": "div",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`People should match the snapshot with no data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Favorites />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="wait"
>
          Access code YT-1300. No favorites found.
</div>,
        false,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Access code YT-1300. No favorites found.",
          "className": "wait",
        },
        "ref": null,
        "rendered": "Access code YT-1300. No favorites found.",
        "type": "div",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="wait"
>
            Access code YT-1300. No favorites found.
</div>,
          false,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Access code YT-1300. No favorites found.",
            "className": "wait",
          },
          "ref": null,
          "rendered": "Access code YT-1300. No favorites found.",
          "type": "div",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
